### PR TITLE
SOLR-16789: Admin UI Dashboard shows "unauthenticated" warning on authenticated servers

### PR DESCRIPTION
**Pull request by tflobbe**
I'm not sure this is the right fix, I don't trust my UI skills. I believe this was introduced in https://github.com/apache/solr/pull/1296

### CODE REVIEW

File: solr/webapp/web/js/angular/controllers/index.js [hunk 1]
```diff
@@ -29,6 +29,10 @@ solrAdminApp.controller('IndexController', function($scope, System, Cores, Const
         sessionStorage.setItem("auth.username", data.security.username);
       }
 
+      if ("authenticationPlugin" in data.security && data.security.authenticationPlugin != null) {
```
Comment (epugh) [inline solr/webapp/web/js/angular/controllers/index.js:32]: I'd have to see if we ahve an established pattern for these if's!   `if data.security.authenticationPlugin ` migth work...

Comment (tflobbe) [reply solr/webapp/web/js/angular/controllers/index.js:32]: Yes, that worked for me

**Review by janhoy — APPROVED** [review]
Have not reproduced or tested the fix, but if it works, it's probably right :)

**Comment by tflobbe** [comment]
Ran these steps:
`gw clean  -p solr/packaging dev && cd  solr/packaging/build/dev`
`bin/solr start -e cloud -noprompt`
Checked that the warning was present in the dashboard
`bin/solr auth enable -credentials foo:bar -z localhost:9983`
On `main`, the warning is still present. With the patch I get:
Screenshot 2023-05-05 at 2 38 27 PM

**Pull request merged**